### PR TITLE
Refactor the boundary conditions code

### PR DESCRIPTION
**Pull request by totto82**
This PR depend on OPM/opm-common#685

### CODE REVIEW

**Comment by totto82** [comment]
jenkins build this opm-common=685 please

**Comment by totto82** [comment]
jenkins build this opm-common=685 please

File: ebos/eclfluxmodule.hh [hunk 1]
```diff
@@ -363,7 +363,7 @@ protected:
     {
         const auto& problem = elemCtx.problem();
 
-        bool enableBoundaryMassFlux = problem.hasFreeBoundaryConditions();
```
Comment (andlaus) [inline ebos/eclfluxmodule.hh]: nit: all decks have boundary conditions, they usually just happen to be trivial (i.e., no-flow)...

Comment (totto82) [reply ebos/eclfluxmodule.hh]: DOne

**Comment by totto82** [comment]
jenkins build this opm-common=685 please

**Pull request merged**